docs/developers/smart-contract-tutorial-foundry.md

```diff
@@ -0,0 +1,208 @@
+# Deploying a Smart Contract to Celestia Ethermint with Foundry
+<!-- markdownlint-disable MD013 -->
+
+In this guide you'll learn how to deploy a Solidity smart contract to Celestia Ethermint with [Foundry](https://github.com/foundry-rs/foundry).
+
+## About Foundry
+
+Foundry is a portable, fast and modular toolkit for Ethereum application development.
+
+Foundry is made up of three components:
+
+[__Forge__](https://github.com/foundry-rs/foundry/tree/master/forge): Ethereum testing framework (like Truffle, Hardhat and DappTools).
+
+[__Cast__](https://github.com/foundry-rs/foundry/tree/master/cast): CLI for interacting with EVM smart contracts, sending transactions, and getting chain data.
+
+[__Anvil__](https://github.com/foundry-rs/foundry/tree/master/anvil): Local Ethereum node, similar to Ganache or Hardhat Network.
+
+We'll use all three to create, test, and deploy our Solidity project.
+
+> To learn more about Foundry, check out the [Foundry Book](https://book.getfoundry.sh/).
+
+## Getting started
+
+To deploy to Celestia Ethermint you will need to request tokens from the test faucet.
+
+### Celestia Ethermint Faucet
+
+> USING THIS FAUCET DOES NOT ENTITLE YOU TO ANY AIRDROP OR OTHER DISTRIBUTION OF CELESTIA OR CELESTIA ETHERMINT TOKENS. MAINNET CELESTIA AND CELESTIA ETHERMINT TOKENS DO NOT CURRENTLY EXIST AND THERE ARE NO PUBLIC SALES OR OTHER PUBLIC DISTRIBUTIONS OF ANY MAINNET CELESTIA OR CELESTIA ETHERMINT TOKENS.
+
+You can request from Celestia Ethermint Faucet on the `#celestia-ethermint-faucet` channel on Celestia's Discord server with the following command:
+
+```sh
+$request <EVM-WALLET-ADDRESS> 
```

Comment:
If this (and other instances of Discord commands) aren't shell commands, then use `text` instead of `sh`? Or something like that.